### PR TITLE
fix(medusa): Received quantity on return lines

### PR DESCRIPTION
**Pull request by olivermrbl**

### CODE REVIEW

**Comment by olivermrbl** [comment]
Changing base to `master` to include in hotfix patch

**Review by kasperkristensen — APPROVED** [review]
LGTM

**Comment by olivermrbl** [comment]
@adrien2p Will you give this a review? Shouldn't take long :)

File: integration-tests/api/__tests__/admin/returns.js [hunk 1]
```diff
@@ -262,4 +263,71 @@ describe("/admin/returns", () => {
...
+    beforeEach(async () => {
+      await adminSeeder(dbConnection)
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      return await db.teardown()
+    })
+
+    it.only("should receive a return partially", async () => {
```
Comment (adrien2p) [inline integration-tests/api/__tests__/admin/returns.js:277]: todo: rm only

File: integration-tests/api/__tests__/admin/returns.js [hunk 2]
```diff
@@ -262,4 +263,71 @@ describe("/admin/returns", () => {
...
+          `/admin/orders/${order.id}/return`,
+          {
+            items: [
+              {
+                item_id: itemId,
+                quantity: 2,
+              },
+            ],
+          },
+          {
```
Comment (adrien2p) [inline integration-tests/api/__tests__/admin/returns.js:295]: todo: Move that to a const at the before the main describe

File: integration-tests/api/__tests__/admin/returns.js [hunk 3]
```diff
@@ -262,4 +263,71 @@ describe("/admin/returns", () => {
...
+            headers: {
+              authorization: "Bearer test_token",
+            },
+          }
+        )
+        .catch((e) => console.log(e))
```
Comment (adrien2p) [inline integration-tests/api/__tests__/admin/returns.js:301]: todo: rm

File: integration-tests/api/__tests__/admin/returns.js [hunk 4]
```diff
@@ -262,4 +263,71 @@ describe("/admin/returns", () => {
...
+          {
+            items: [
+              {
+                item_id: itemId,
+                quantity: 1,
+              },
+            ],
+          },
+          {
+            headers: {
```
Comment (adrien2p) [inline integration-tests/api/__tests__/admin/returns.js:318]: todo: use the const mentionned before

File: integration-tests/api/__tests__/admin/returns.js [hunk 5]
```diff
@@ -262,4 +263,71 @@ describe("/admin/returns", () => {
...
+              authorization: "Bearer test_token",
+            },
+          }
+        )
+        .catch((e) => console.log(e))
```
Comment (adrien2p) [inline integration-tests/api/__tests__/admin/returns.js:323]: todo: rm

File: integration-tests/api/__tests__/admin/returns.js [hunk 6]
```diff
@@ -262,4 +263,71 @@ describe("/admin/returns", () => {
...
+
+      const receivedReturn = receiveRes.data.return
+
+      const receivedItem = receivedReturn.items[0]
```
Comment (adrien2p) [inline integration-tests/api/__tests__/admin/returns.js:327]: suggestion: should we expect the length?

**Review by adrien2p — CHANGES_REQUESTED** [review]
few changes in the test and it should be good to go 🚀

File: integration-tests/api/__tests__/admin/returns.js [hunk 7]
```diff
@@ -274,60 +234,49 @@ describe("/admin/returns", () => {
...
+        authHeader
+      )
 
       const receivedReturn = receiveRes.data.return
 
       const receivedItem = receivedReturn.items[0]
 
       expect(receivedItem.requested_quantity).toEqual(2)
       expect(receivedItem.received_quantity).toEqual(1)
+      expect(receivedReturn.items.length).toEqual(1)
```
Comment (adrien2p) [inline integration-tests/api/__tests__/admin/returns.js:279]: todo: this expect should happen before line 275 to ensure we get the right one

**Review by adrien2p — APPROVED** [review]
One last change, LGTM

**Pull request merged**